### PR TITLE
Adding latest fixes, tests are passing.

### PR DESCRIPTION
**Pull request by manderiasian**

### CODE REVIEW

File: src/libseshat.c [hunk 1]
```diff
@@ -223,13 +223,15 @@ bool send_message(int wrp_request, const char *service,
...
-            msg->u.crud.path = (char *) service;
-            msg->u.crud.transaction_uuid = uuid;
+            msg->u.crud.path = strdup(service);
+            msg->u.crud.transaction_uuid = strdup(uuid);
  /* ??? If source/dest are NULL then wrp_struct_to() works however
   *     the resulting byte array fails with wrp_to_struct()
  */  
-            msg->u.crud.source = ""; 
-            msg->u.crud.dest = "";   
+            msg->u.crud.source  = strdup("Bazinga");
```
Comment (schmidtw) [inline src/libseshat.c]: The response message's source should be the request message's dest.  The response message's dest should be the request message's src.

Comment (manderiasian) [inline src/libseshat.c]: Ummm....the current libseshat API, i.e. "seshat_discover(char * service)" does not have a source/dest, and in the above this is a message being build based on that call. So, where do we get source/dest for a request? I can add that so we would have "seshat_discover(service, source, dest)" is that what u r thinking?

Comment (manderiasian) [inline src/libseshat.c]: BTW In wrp_interface.c, it does swap the fields when responding to a [R] request.

File: src/libseshat.c [hunk 2]
```diff
@@ -213,8 +223,15 @@ bool send_message(int wrp_request, const char *service,
...
     switch (wrp_request) {
         case WRP_MSG_TYPE__RETREIVE:
-            msg->u.crud.path = (char *) service;
+            msg->u.crud.path = strdup(service);
             msg->u.crud.transaction_uuid = strdup(uuid);
+ /* ??? If source/dest are NULL then wrp_struct_to() works however
+  *     the resulting byte array fails with wrp_to_struct()
+ */  
+            msg->u.crud.source  = strdup("Bazinga");
+            msg->u.crud.dest    = strdup("Agnizab");
```
Comment (manderiasian) [inline src/libseshat.c]: I already replied to Wes. The code does not have sixth sense ;-), where is it supposed to get those? I guess dest could be "seshat-server" hard coded, but what is the source? libseshat has no idea who the client is.

**Pull request merged**